.github/workflows/e2e-tests.yml

```diff
@@ -194,26 +194,26 @@ jobs:
               patch -b data/TraitOne.php < TraitOne.patch
               OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
               echo "$OUTPUT"
-              [ $(echo "$OUTPUT" | wc -l) -eq 1 ]
-              grep 'Method TraitsCachingIssue\\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
+              ../bashunit -a line_count 1 "$OUTPUT"
+              ../bashunit -a contains 'Method TraitsCachingIssue\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' "$OUTPUT"
           - script: |
               cd e2e/trait-caching
               ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
               patch -b data/TraitTwo.php < TraitTwo.patch
               OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
               echo "$OUTPUT"
-              [ $(echo "$OUTPUT" | wc -l) -eq 1 ]
-              grep 'Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
+              ../bashunit -a line_count 1 "$OUTPUT"
+              ../bashunit -a contains 'Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.' "$OUTPUT"
           - script: |
               cd e2e/trait-caching
               ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
               patch -b data/TraitOne.php < TraitOne.patch
               patch -b data/TraitTwo.php < TraitTwo.patch
               OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
               echo "$OUTPUT"
-              [ $(echo "$OUTPUT" | wc -l) -eq 2 ]
-              grep 'Method TraitsCachingIssue\\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
-              grep 'Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
+              ../bashunit -a line_count 2 "$OUTPUT"
```

Comment:
I would like to reproduce it locally, so I cloned your fork (`git@github.com:Chemaclass/staabm-phpstan-src.git`), checkout your bashunit branch, and I get this error when running composer install - which I think it shouldn't happen as it is a virgin start (I also set php 8.1), so I guess there is a problem on the deps? Did you face the same issue, @staabm ? Any idea how to find a work around to this?  ![340425921-5b9fabbb-1df8-4442-bddc-29454deabaae](https://github.com/phpstan/phpstan-src/assets/5256287/a5dc6608-6100-4bfb-842e-74bf6fa2514f)

Reply:
See the README.

> If you encounter dependency problem, try using `export COMPOSER_ROOT_VERSION=1.11.x-dev`

Reply:
I think I can reproduce on the CLI outside of phpstan repo with:

```
OUTPUT="one\ntwo\three"
./bashunit -a line_count 3 "$OUTPUT"
```

Reply:
Ok, in such a case the bug seems to be in the `assert_line_count` where it might consider that as 1 line and not as 3. I think that's the direction of the 🐛

Reply:
hmm .. the unit test regarding it looked good:
https://github.com/TypedDevs/bashunit/pull/259/files#diff-8c5ca8e706f9d9fb330cda7953d707bb4180b68f55fdff87ec30ad3f8e179ad6R339-R341

I don't know bash good enough, but I had a feeling the problem is related to the assertion via args passing

Reply:
The issue is clearly here `actual_line_count=$(echo "$actual" | wc -l | tr -d '[:blank:]')` 
Here you can see the result of this test:
```bash
function test_successful_assert_line_count_multiline_string_in_one_line() {
  assert_empty "$(assert_line_count 4 "one\ntwo\nthree\nfour")"
}
```
![Screenshot 2024-06-17 at 22 50 33](https://github.com/phpstan/phpstan-src/assets/5256287/24efb3ca-e066-41b2-a56c-f9273b05b4c2)

Reply:
I got a potential fix for this ... creating PR

Reply:
I got a potential fix for this: https://github.com/TypedDevs/bashunit/pull/264 (merged)
Can you try again to reproduce it with the latest beta, @staabm ?

Reply:
I can reproduce it, yes. This is an issue, I will work on it today 🏗️

Reply:
Alright, https://github.com/TypedDevs/bashunit/pull/265 is merged and we are getting closer to the light!
Can you try again to reproduce it with the latest beta, @staabm ?

Reply:
With the latest [error](https://github.com/phpstan/phpstan-src/actions/runs/9567238902/job/26374488151), I think we might have found a major problem wrt multiline strings mmmm I have an idea, but it might take a while... let me see if I can create a draft for this today 🏗️

Reply:
Same drill: https://github.com/TypedDevs/bashunit/pull/266 is merged, can you try the latest beta again, @staabm? 🤞🏼

Reply:
@Chemaclass all e2e are green now. thank you very much for your efforts <3.